### PR TITLE
Added basic F# notebook for benchmarking

### PR DESCRIPTION
**Pull request by TIHan**
This does some cleanup to FSharp.Compiler.Benchmarks and adds a basic .NET Interactive notebook that will run a F# benchmark and display the GC collection counts in a graph. This is the first of many benchmarks we will make.

benchmark1_test

### CODE REVIEW

File: tests/benchmarks/CompilerServiceBenchmarks/Benchmarks.fs [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+open FSharp.Compiler.AbstractIL.IL
+open FSharp.Compiler.AbstractIL.ILBinaryReader
+open BenchmarkDotNet.Attributes
+open BenchmarkDotNet.Running
+open FSharp.Compiler.Benchmarks
+
+[<AutoOpen>]
+module BenchmarkHelpers =
+
+    type Async with
```
Comment (vzarytovskii) [inline tests/benchmarks/CompilerServiceBenchmarks/Benchmarks.fs:20]: Should we reuse it from the test utilities?
https://github.com/dotnet/fsharp/blob/fca1cd7bb08f01aa7abd34876d6ee748ec5fc8e9/tests/FSharp.Test.Utilities/Utilities.fs#L21

File: tests/benchmarks/CompilerServiceBenchmarks/Benchmarks.fs [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+        static member RunImmediate (computation: Async<'T>, ?cancellationToken ) =
```
Comment (vzarytovskii) [inline tests/benchmarks/CompilerServiceBenchmarks/Benchmarks.fs:21]: Should we reuse it from the test utilities?
https://github.com/dotnet/fsharp/blob/fca1cd7bb08f01aa7abd34876d6ee748ec5fc8e9/tests/FSharp.Test.Utilities/Utilities.fs#L21

**Review by vzarytovskii — APPROVED** [review]

File: tests/benchmarks/CompilerServiceBenchmarks/Benchmarks.fs (hunk 2 above)
Comment (TIHan) [reply tests/benchmarks/CompilerServiceBenchmarks/Benchmarks.fs:21]: Would be nice, but I think it's fine to have duplicates as we may not want to have extra dependencies

**Comment by kerams** [comment]
It would be cool to have this computed and published on a daily basis, so that potential regressions are easier to spot, just like they do for parts of BCL

**Comment by vzarytovskii** [comment]
> It would be cool to have this computed and published on a daily/regular basis, so that potential regressions are easier to spot, just like they do for parts of BCL
> 
> 
> 
> 

Yeah, we need to utilise our perf infra to have consistent numbers across the runs. That would be nice!

**Pull request merged**